src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -19,8 +19,7 @@ def __init__(self, token_retriever, external_tenant_token_retriever=None):
         self._token_retriever = token_retriever
         self._external_tenant_token_retriever = external_tenant_token_retriever
 
-    def signed_session(self, session=None):  # pylint: disable=arguments-differ
-        session = session or super(AdalAuthentication, self).signed_session()
+    def fetch_tokens(self):
```

Comment:
There is [get_raw_token](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/_profile.py#L587) which might be enough, so that you don't need to refactor here.

Reply:
I wasn't aware of that one. I will use that instead of changing the AdalAuthentication object.

Reply:
I reverts the changes to AdalAuthentication and used get_raw_token as you requested. Let me know if you need anything else here.